### PR TITLE
Revamp Android deployment docs

### PR DESCRIPTION
**Pull request by lazerwalker**
#### Changes
  
This revamps the Android GitHub Actions deployment docs to be more in-depth and to not assume familiarity with Fastlane or Play Store deployment, in line with the new iOS deployment docs.

#### Checklist

- [x] Read the contribution [guide](../blob/main/CONTRIBUTING.md) and accept the [code](../blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Review by webbertakken — APPROVED** [review]

File: docs/github/v2/15-deployment/android.md [hunk 1]
```diff
@@ -1,32 +1,72 @@
...
 
-The recommended approach to install [fastlane](https://docs.fastlane.tools/getting-started/android/setup/)
-is to make a `Gemfile` with following content:
+### 1- Install Fastlane
+
+[Fastlane](https://docs.fastlane.tools/getting-started/ios/setup/) is a tool that can facilitate building and submitting your Android apps to Google, and is the easiest way to deploy your Unity project to Android for Play Store distribution.
+
+To configure Fastlane for your GitHub Actions workflow runners, you will need to locally set up a `Gemfile` and `Fastfile` within your project. A `Gemfile` specifies what Ruby dependencies are needed to set up and run Fastlane (which is written in Ruby), and a `Fastfile` will be how you configure your iOS build settings. We will set up the `Gemfile` now, and the `Fastfile` in a later step.
```
Comment (davidmfinol) [inline docs/github/v2/15-deployment/android.md:9]:
```suggestion
To configure Fastlane for your GitHub Actions workflow runners, you will need to locally set up a `Gemfile` and `Fastfile` within your project. A `Gemfile` specifies what Ruby dependencies are needed to set up and run Fastlane (which is written in Ruby), and a `Fastfile` will be how you configure your Android deployment settings. We will set up the `Gemfile` now, and the `Fastfile` in a later step.
```

File: docs/github/v2/15-deployment/android.md [hunk 2]
```diff
@@ -1,32 +1,72 @@
...
+
+### 2- Create a Google Play Service Account
+
+To programmatically access the Google Play Console, you will need a dedicated Google Play service account with API access.
+
+Follow the ["Setup" section of the Fastlane Supply documentation](https://docs.fastlane.tools/actions/supply/) to create a service account.
+
+After the last step, it will tell you to test the connection to the Google Play Store, and then add your JSON file path to your Appfile. You can do the first step if you would like (you may need to run `bundle exec fastlane` instead of just `fastlane`), but don't worry about the Appfile, as we will be be creating that in the next step.
+
+Instead, create a Repository Secret in your GitHub repository by going to Settings -> Secrets and clicking the "New repository secret" button in the top-right. It should be titled `GOOGLE_PLAY_KEY_FILE` and its value should be the plaintext contents of the download JSON file.
```
Comment (davidmfinol) [inline docs/github/v2/15-deployment/android.md:37]:
```suggestion
Instead, create a Repository Secret in your GitHub repository by going to Settings -> Secrets and clicking the "New repository secret" button in the top-right. It should be titled `GOOGLE_PLAY_KEY_FILE` and its value should be the plaintext contents of the downloaded JSON file.
```

File: docs/github/v2/15-deployment/android.md [hunk 3]
```diff
@@ -1,32 +1,72 @@
 # Deploy to Google Play
 
-### 1- Install fastlane
+This guide is intended to help with automating Android builds to upload to the Play Store. If you are planning to use an alternate distribution method, such as a third-party beta test service or manually sideloading an APK, you do not need this guide. Setting up the [unity-builder](https://game.ci/docs/github/builder) action with a `targetPlatform` of `Android` should produce a functioning APK. However, if you are intending to distribute your game on the Play Store, either for public distribution or a beta track, this guide is for you.
```
[the remark this answers is not in the document]
Comment (lazerwalker) [reply docs/github/v2/15-deployment/android.md:3]: Good catch, I agree this paragraph should be a bit more concise! Going to spend a little more time thinking about how to best condense it.

File: docs/github/v2/15-deployment/android.md [hunk 4]
```diff
@@ -39,17 +79,25 @@ end
...
+
+### 6- Add jobs to your GitHub workflow
+
+The following workflow establishes two jobs. The first builds your game into an AAB file, and the second uploads that generated bundle to the Play Store.
+
+If you instead intend to manually distribute your game outside the Play Store as an APK, remove the `androidAppBundle` flag from the `with` section of the `game-ci/unity-builder@v2` action in the workflow. You can also remove the `androidKeystoreName`, `androidKeystoreBase64`, `androidKeyStorePass`, `androidKeyaliasName`, and `androidKeyaliasPass` properties. From there, you'll want to remove the entire `releaseToGooglePlay` job. If you just want to download the APK, it will be available in the "artifacts" section of a completed workflow run, or you can add a new job that performs whatever custom upload logic you'd like.
```
Comment (davidmfinol) [inline docs/github/v2/15-deployment/android.md:100]: I don't think we need this paragraph, as it has already been established that this guide is specifically for deployment to the Google Play Store.

Comment (lazerwalker) [reply docs/github/v2/15-deployment/android.md]: Totally agree, removed

File: docs/github/v2/15-deployment/android.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (lazerwalker) [reply docs/github/v2/15-deployment/android.md]: This is now shorter!

**Comment by lazerwalker** [comment]
I think I'm now happy with the state of where this is (great suggestions, everyone!) but I'm naturally open to more feedback.

**Review by webbertakken — APPROVED** [review]

**Pull request merged**